### PR TITLE
[Feature] Add nigerian provider

### PR DESCRIPTION
**Pull request by andela-celisha-wigwe**
This PR adds a Nigerian provider to the faker project.

Things added include:

- Nigerian States, cities and regions
- Nigerian names, both male and female
- Phone number formats
- Internet formats

Unit Tests are also written to ensure expected results are achieved.

Set your application to use the provider by passing `ng_NG` to the create static method. Thus  

`$faker = Faker\Factory::create('ng_NG');`

### CODE REVIEW

File: src/Faker/Provider/ng_NG/Address.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        'Plateau',
+        'Rivers',
+        'Sokoto',
+        'Taraba',
+        'Yobe',
+        'Zamfara'
+    );
+
+    protected static $country = array(
+        'Afghanistan', 'Albania', 'Algeria', 'American Samoa', 'Andorra', 'Angola', 'Anguilla', 'Antarctica (the territory South of 60 deg S)', 'Antigua and Barbuda', 'Argentina', 'Armenia', 'Aruba', 'Australia', 'Austria', 'Azerbaijan',
```
Comment (fzaninotto) [inline src/Faker/Provider/ng_NG/Address.php]: This is English. Isn't there a Hausa / Igbo / Yoruba version?

File: src/Faker/Provider/ng_NG/Person.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    );
+
+    protected static $femaleNameFormats = array(
+        '{{firstNameFemale}} {{lastName}}',
+        '{{firstNameFemale}} {{firstNameFemale}} {{lastName}}'
+    );
+
+    /**
+     * An array of typical Nigerian male firstnames.
+     * @link http://www.top-100-baby-names-search.com/baby-names-nigeria.html
```
Comment (fzaninotto) [inline src/Faker/Provider/ng_NG/Person.php]: This content is copyrighted, please use data from a MIT-compatible source

File: src/Faker/Provider/ng_NG/Person.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        'Remilekun', 'Rotimi',
+        'Shade', 'Shalewa', 'Sname',
+        'Tari', 'Temitope', 'Titilope', 'Tobiloba', 'Toke', 'Tomiloba', 'Tope',
+        'Uzodimma',
+        'Wale',
+        'Yakubu', 'Yusuf', 'Yusuf',
+    );
+
+    /**
+     * An array of typical Nigerian female firstnames.
```
Comment (fzaninotto) [inline src/Faker/Provider/ng_NG/Person.php]: Missing datasource

File: src/Faker/Provider/ng_NG/Person.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        'Lola',
+        'Obioma', 'Omawunmi', 'Omolara', 'Onome',
+        'Rasheedah',
+        'Sekinat', 'Simisola', 'Sumayyah',
+        'Titi', 'Titilayo', 'Toluwani',
+        'Zainab'
+    );
+
+    /**
+     * An array of typical Nigerian lastnames.
```
Comment (fzaninotto) [inline src/Faker/Provider/ng_NG/Person.php]: Missing datasource

File: test/Faker/Provider/ng_NG/AddressTest.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Faker\Provider\ng_NG;
+
+use Faker\Generator;
+use Faker\Provider\ng_NG\Address;
+use Faker\Test\Provider\ng_NG\MainTest;
+
+class AddressTest extends MainTest
```
Comment (fzaninotto) [inline test/Faker/Provider/ng_NG/AddressTest.php]: no, please extend `\PHPUnit_Framework_TestCase` instead. `performTest()` doesn't mean anything.

File: test/Faker/Provider/ng_NG/AddressTest.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    {
+        $faker = new Generator();
+        $faker->addProvider(new Address($faker));
+        $this->faker = $faker;
+    }
+
+    /**
+     *
+     */
+    public function testPostcode()
```
Comment (fzaninotto) [inline test/Faker/Provider/ng_NG/AddressTest.php]: You'll need to test at least 2 things:

- `testPostCodeIsntEmpty()`
- `testPostCodeReturnsAValidPostCode()`

Same for the other tests

File: test/Faker/Provider/ng_NG/MainTest.php [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+     */
+    public function performTest($param)
+    {
+        $this->assertNotEmpty($param);
+        $this->assertInternalType('string', $param);
+    }
+
+    public function testTrue()
+    {
+        // $this->performTest($this->faker->region);
```
Comment (fzaninotto) [inline test/Faker/Provider/ng_NG/MainTest.php]: commented out code

File: test/Faker/Provider/ng_NG/MainTest.php [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        $faker = new Generator();
+        $faker->addProvider(new Address($faker));
+        $this->faker = $faker;
+    }
+
+    /**
+     * Test that the given parameter is not empty
+     * And is also a string
+     */
+    public function performTest($param)
```
Comment (fzaninotto) [inline test/Faker/Provider/ng_NG/MainTest.php]: very bad naming, too broad, please remove

File: test/Faker/Provider/ng_NG/PersonTest.php [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    private $faker;
+
+    public function setUp()
+    {
+        $faker = new Generator();
+        $faker->addProvider(new Person($faker));
+        $this->faker = $faker;
+    }
+
+    public function testFirstName()
```
Comment (fzaninotto) [inline test/Faker/Provider/ng_NG/PersonTest.php]: `testFirstName` what? The test title should mention what you test, contain a verb

File: test/Faker/Provider/ng_NG/PhoneNumberTest.php [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+class PhoneNumberTest extends MainTest
+{
+    public function setUp()
+    {
+        $faker = new Generator();
+        $faker->addProvider(new PhoneNumber($faker));
+        $this->faker = $faker;
+    }
+
+    public function testPhoneNumberReturnsPhoneNumberWithOrWithoutPrefix()
```
Comment (fzaninotto) [inline test/Faker/Provider/ng_NG/PhoneNumberTest.php]: this one is better

File: src/Faker/Provider/ng_NG/Address.php (hunk 1 above)
Comment (andela-celisha-wigwe) [inline src/Faker/Provider/ng_NG/Address.php]: You are right. However, the English names of countries are very commonly used in Nigeria. We do not really have localised names.

File: src/Faker/Provider/ng_NG/Person.php (hunk 3 above)
Comment (andela-celisha-wigwe) [inline src/Faker/Provider/ng_NG/Person.php]: The source is the same for all the names. However, I would include it.

File: src/Faker/Provider/ng_NG/Address.php (hunk 1 above)
Comment (Ema4rl) [inline src/Faker/Provider/ng_NG/Address.php]: That's right.

File: src/Faker/Provider/ng_NG/Person.php (hunk 2 above)
Comment (andela-celisha-wigwe) [inline src/Faker/Provider/ng_NG/Person.php]: I could not find an open source link for Nigerian names. So I created a little one, with an aim to expand it in the future.

File: src/Faker/Provider/ng_NG/Person.php [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+    );
+
+    protected static $femaleNameFormats = array(
+        '{{firstNameFemale}} {{lastName}}',
+        '{{firstNameFemale}} {{firstNameFemale}} {{lastName}}'
+    );
+
+    /**
+     * An array of typical Nigerian male firstnames.
+     * @link https://andela-celisha-wigwe.github.io/names.html
```
Comment (fzaninotto) [inline src/Faker/Provider/ng_NG/Person.php]: There is no open-source License [on your content](https://github.com/andela-celisha-wigwe/andela-celisha-wigwe.github.io). Faker can only reuse it if the content uses MIT or a compatible license.

File: test/Faker/Provider/ng_NG/InternetTest.php [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        $faker->addProvider(new Internet($faker));
+        $this->faker = $faker;
+    }
+
+    public function testEmailIsValid()
+    {
+        $email = $this->faker->email();
+        $this->assertNotFalse(filter_var($email, FILTER_VALIDATE_EMAIL));
+        $this->assertNotEmpty($email);
+        $this->assertInternalType('string', $email);
```
Comment (fzaninotto) [inline test/Faker/Provider/ng_NG/InternetTest.php]: You should add an [email validator}(http://php.net/manual/en/filter.examples.validation.php).

File: src/Faker/Provider/ng_NG/Address.php [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+        'Taiwan', 'Tajikistan', 'Tanzania', 'Thailand', 'Timor-Leste', 'Togo', 'Tokelau', 'Tonga', 'Trinidad and Tobago', 'Tunisia', 'Turkey', 'Turkmenistan', 'Turks and Caicos Islands', 'Tuvalu',
+        'Uganda', 'Ukraine', 'United Arab Emirates', 'United Kingdom', 'United States of America', 'United States Minor Outlying Islands', 'United States Virgin Islands', 'Uruguay', 'Uzbekistan',
+        'Vanuatu', 'Venezuela', 'Vietnam',
+        'Wallis and Futuna', 'Western Sahara',
+        'Yemen',
+        'Zambia', 'Zimbabwe'
+    );
+
+    private static $regions = array( 'Umuahia', 'Yola', 'Uyo', 'Awka', 'Bauchi', 'Yenagoa', 'Makurdi', 'Maiduguri', 'Calabar', 'Asaba', 'Abakaliki', 'City', 'Ekiti', 'Enugu', 'Gombe', 'Owerri', 'Dutse', 'Kaduna', 'Kano', 'Katsina', 'Kebbi', 'Lokoja', 'Ilorin', 'Ikeja', 'Lafia', 'Minna', 'Abeokuta', 'Akure', 'Oshogbo', 'Ibadan', 'Jos', 'Harcourt', 'Sokoto', 'Jalingo', 'Damaturu', 'Gusau'
+    );
```
Comment (fzaninotto) [inline src/Faker/Provider/ng_NG/Address.php]: CS: if you inline the array, you should also inline the closing parenthesis. And please order the region list alphabetically for easier debugging.

File: test/Faker/Provider/ng_NG/InternetTest.php (hunk 12 above)
Comment (andela-celisha-wigwe) [inline test/Faker/Provider/ng_NG/InternetTest.php]: I believe you are referring to Line 28
```php
$this->assertNotFalse(filter_var($email, FILTER_VALIDATE_EMAIL));
```

Comment (fzaninotto) [inline test/Faker/Provider/ng_NG/InternetTest.php]: My bad, I didn't see it.

**Comment by fzaninotto** [comment]
Awesome, thanks for your efforts!

**Comment by andela-celisha-wigwe** [comment]
It's always a good thing relating with more experienced developers. I've learnt a couple of things from this. Thanks for merging. 👍

**Pull request merged**